bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/utils/DbMetaData.java

```diff
@@ -0,0 +1,117 @@
+package org.openhab.persistence.jdbc.utils;
+
+import java.sql.DatabaseMetaData;
+import java.sql.SQLException;
+
+import org.knowm.yank.Yank;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.zaxxer.hikari.HikariDataSource;
+
+/**
+ *
+ * @author Helmut Lehmeyer
+ * @since 1.8.0
+ */
+public class DbMetaData {
+
+    private static final Logger logger = LoggerFactory.getLogger(DbMetaData.class);
+
+    private int dbMajorVersion;
+    private int dbMinorVersion;
+    private int driverMajorVersion;
+    private int driverMinorVersion;
+    private String dbProductName;
+    private String dbProductVersion;
+
+    public DbMetaData() {
+        HikariDataSource h = Yank.getDefaultConnectionPool();
+        // HikariDataSource h = Yank.getDataSource();
+
+        DatabaseMetaData meta;
+        try {
+            meta = h.getConnection().getMetaData();
+
+            // Oracle (and some other vendors) do not support
+            // some the following methods; therefore, we need
+            // to use try-catch block.
+            try {
+                dbMajorVersion = meta.getDatabaseMajorVersion();
+                logger.debug("dbMajorVersion = '{}'", dbMajorVersion);
+            } catch (Exception e) {
+                logger.error("Asking for 'dbMajorVersion' is a unsupported: '{}'", e);
+            }
+
+            try {
+                dbMinorVersion = meta.getDatabaseMinorVersion();
+                logger.debug("dbMinorVersion = '{}'", dbMinorVersion);
+            } catch (Exception e) {
+                logger.error("Asking for 'dbMajorVersion' is a unsupported: '{}'", e);
+            }
+
+            driverMajorVersion = meta.getDriverMajorVersion();
+            logger.debug("driverMajorVersion = '{}'", driverMajorVersion);
+
+            driverMinorVersion = meta.getDriverMinorVersion();
+            logger.debug("driverMinorVersion = '{}'", driverMinorVersion);
+
+            dbProductName = meta.getDatabaseProductName();
+            logger.debug("dbProductName = '{}'", dbProductName);
+
+            dbProductVersion = meta.getDatabaseProductVersion();
+            logger.debug("dbProductVersion = '{}'", dbProductVersion);
+
+        } catch (SQLException e1) {
+            logger.error("Asking for 'dbMajorVersion' seams to be unsupported: '{}'", e1);
```

Comment:
Spelling: seems

Also, could the exception be from a different method call on the `meta` object, so maybe the exception message might have the true answer, so you don't need to guess?